### DISCUSSION TITLE
Weird username link

### DISCUSSION THREAD

**Discussion by kissdry** [community/community#29903] [Other Feature Feedback, Questions, & Ideas]
Not sure, if it is a proper way to ask this question, but here it is.

1) here is my account: https://github.com/kissdry

2) Have no idea where I set it, but my "user nickname" (or whatever it is) is "Jim", and I totally comfortable with it
![Screenshot from 2022-08-18 10-37-35](https://user-images.githubusercontent.com/22590409/185337656-a7182d75-a845-4aa4-bed5-b6d49088db9f.png)

3) But somehow GitHub thinks that another person (or is it still me?) is making all commits into my repos (someone named "JimBerg")
![Screenshot from 2022-08-18 10-39-26](https://user-images.githubusercontent.com/22590409/185337939-56558d08-bfbb-4aad-bf8d-2c2b893a9d91.png)

There is a chance that I don't see something obvious, but I used to be nicknamed as "Jim" for the last 25 years and the fact that I can not get rid of this "JimBerg" just drives me mad :)

**Comment by radar-blip** [comment]
So if you think theres a possibility that it might still be you, what makes you think it isn't?

Here is the JimBerg account. https://github.com/JimBerg  Is it you?

**Reply by kissdry** [reply]
As I have said, I have never ever heard of "JimBerg", never wrote this combination of letters anywhere etc. I have no idea where on github I can change this nickname. And no, I have no relation to the URL provided.

**Reply by radar-blip** [reply]
Have you go this sorted out yet? 

You can change your name 'Jim' to whatever you want. You can change the username also, but it might have unwanted side-effects.
This is all in your settings.

**Reply by kissdry** [reply]
No, I have not.
I am absolutely ok with both "Jim" and "kissdry".
I am not even against "JimBerg" as far as it has no impact neither on me nor on my repos.
I just want to understand what "JimBerg" is and where it came from.
And, of course, would be great to change it to "jim" as it shoud be from my point of view.

**Reply by radar-blip** [reply]
Ok well, if you scroll down the page further, the last comment there gets to the point of the issue (i think).
You may want to read it, if you havent already.

**Comment by ChewyToast** [comment]
This JimBerg name is probably the name set in your local .gitprofile, as well as the name of all the commits you're doing. About he Jim name u can find it ont ur github settings

**Reply by ChewyToast** [reply]
if u just want to change it use this commands:
- git config [--global] user.name "Full Name"
- git config [--global] user.email "email@address.com"

**Reply by kissdry** [reply]
No, it is not!

git config --list
user.name=jim
user.email=jim@home
core.autocrlf=input
core.repositoryformatversion=0
core.filemode=true
core.bare=false
core.logallrefupdates=true
remote.origin.url=git@github.com:kissdry/yii2-messages.git
remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
branch.master.remote=origin
branch.master.merge=refs/heads/master

git log
commit f466a8330bd3c98149dd55f3ae722e3631ca432b (HEAD -> master, origin/master)
Author: jim <jim@home>
Date:   Thu Aug 18 07:54:56 2022 +0300

    Remove testing readme

commit abd750a49ea852b13d32aeb9b5806f3258ec458f
Author: jim <jim@home>
Date:   Thu Aug 18 07:47:26 2022 +0300

    Added web service and html interface

commit 8c7a378dbefde237229fdef080a5ac0a734694c9
Author: jim <jim@home>
Date:   Wed Aug 17 14:21:16 2022 +0300

    Added data generation console command

**Reply by ChewyToast** [reply]
waw, im not sure how can this happen i can only say to you to check on ur profile settings the emails sets, just to check.

**Reply by kissdry** [reply]
I don't know where else I can check/change it. There are billions of params on github, but I can't find what I need. And it's more and more seems to be a weird bug :)

![image](https://user-images.githubusercontent.com/22590409/185374752-f89ae3c8-b87b-40ba-bfce-7af8f852cee8.png)

**Comment by airtower-luna** [comment]
There's your problem:

> git config --list
> user.name=jim
> user.email=jim@home

The `user.email` is what ends up in your commit as the author email. And the author email is what GitHub uses to assign commits to user accounts. I assume the "JimBerg" user made the same mistake and then just added the invalid address to their profile. :sweat_smile: 

To get your commits correctly assigned to your account, the email address must be one that's on your account, or the [GitHub-provided no-reply address](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address#about-commit-email-addresses) (if you want to keep your actual mail address private).

To fix past commits you'd need to [rewrite the repository history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History). For a repository you're using on your own it's not a big deal, but if it's shared with others changing history is likely to cause major problems.

**Reply by radar-blip** [reply]
So, if OP changes his email address, JimBerg disappears?

**Reply by airtower-luna** [reply]
At least for future commits, yes. :slightly_smiling_face: 

Whether it can be fixed for past commits depends on whether changing the history is reasonably possible.

**Reply by radar-blip** [reply]
Ok. Thanks for clearing things up! 👍🏻
